crates/cast/bin/cmd/tx_pool.rs

```diff
@@ -0,0 +1,90 @@
+use alloy_primitives::Address;
+use alloy_provider::Provider;
+use alloy_rpc_types_txpool::{TxpoolContent, TxpoolContentFrom, TxpoolInspect, TxpoolStatus};
+use clap::Parser;
+use foundry_cli::{
+    opts::RpcOpts,
+    utils::{self, LoadConfig},
+};
+
+/// CLI arguments for `cast tx-pool`.
+#[derive(Debug, Parser, Clone)]
+pub enum TxPoolSubcommands {
+    Content {
+        #[command(flatten)]
+        args: RpcOpts,
+    },
+    ContentFrom {
+        /// The Signer to filter the transactions by.
+        #[arg(short, long)]
+        from: Address,
+        #[command(flatten)]
+        args: RpcOpts,
+    },
+    Inspect {
+        #[command(flatten)]
+        args: RpcOpts,
+    },
+    Status {
+        #[command(flatten)]
+        args: RpcOpts,
+    },
+}
+
+impl TxPoolSubcommands {
+    pub async fn run(self) -> eyre::Result<()> {
+        match self {
+            TxPoolSubcommands::Content { args } => {
+                let config = args.load_config()?;
+                let provider = utils::get_provider(&config)?;
+
+                sh_println!(
+                    "{:#?}",
+                    provider
+                        .client()
+                        .request::<_, TxpoolContent>("txpool_content", "")
+                        .boxed()
+                        .await?
+                )?;
+            }
+            TxPoolSubcommands::ContentFrom { from, args } => {
+                let config = args.load_config()?;
+                let provider = utils::get_provider(&config)?;
+                sh_println!(
+                    "{:#?}",
+                    provider
+                        .client()
+                        .request::<_, TxpoolContentFrom>("txpool_contentFrom", [from])
+                        .boxed()
+                        .await?
+                )?;
+            }
+            TxPoolSubcommands::Inspect { args } => {
```

Comment:
This is super useful for mempool inspection.
To make it better, good to extend this to support:
- Filtering by nonce/gasPrice
- Output formatting (pretty vs raw)
- Export to JSON file for offline inspection